security-analysis/security-analysis-api/src/main/java/com/powsybl/security/LimitViolationHelper.java

```diff
@@ -24,6 +24,14 @@ private static VoltageLevel getVoltageLevel(LimitViolation limitViolation, Netwo
         Objects.requireNonNull(limitViolation);
 
         Identifiable identifiable = network.getIdentifiable(limitViolation.getSubjectId());
+        if (limitViolation.getLimitType() == LimitViolationType.LOW_VOLTAGE_ANGLE || limitViolation.getLimitType() == LimitViolationType.HIGH_VOLTAGE_ANGLE) {
+            Optional<VoltageAngleLimit> limit = network.getVoltageAngleLimits().stream().filter(l -> l.getName().equals(limitViolation.getSubjectId())).findAny();
```

Comment:
As far as name would be come and ID and unicity is ensured we could add a network.getVoltageAngleLimit(id) more efficient